03_Modules/Transactions/src/module/module.ts

```diff
@@ -412,6 +305,127 @@ export class TransactionsModule extends AbstractModule {
+                    0
+                    ? (authorization.idTokenInfo.groupIdToken.additionalInfo.map(
+                        (additionalInfo) => ({
+                          additionalIdToken: additionalInfo.additionalIdToken,
+                          type: additionalInfo.type,
+                        }),
+                      ) as [AdditionalInfoType, ...AdditionalInfoType[]])
+                    : undefined,
+                idToken: authorization.idTokenInfo.groupIdToken.idToken,
+                type: authorization.idTokenInfo.groupIdToken.type,
+              }
+            : undefined,
+          language2: authorization.idTokenInfo.language2,
+          personalMessage: authorization.idTokenInfo.personalMessage,
+        };
+
+        if (idTokenInfo.status === AuthorizationStatusEnumType.Accepted) {
+          if (
+            idTokenInfo.cacheExpiryDateTime &&
+            new Date() > new Date(idTokenInfo.cacheExpiryDateTime)
+          ) {
+            transactionEventResponse.idTokenInfo = {
+              status: AuthorizationStatusEnumType.Invalid,
+              groupIdToken: idTokenInfo.groupIdToken,
+              // TODO determine how/if to set personalMessage
+            };
+          } else {
+            // TODO: Determine how to check for NotAllowedTypeEVSE, NotAtThisLocation, NotAtThisTime, NoCredit
+            // TODO: allow for a 'real time auth' type call to fetch token status.
+            transactionEventResponse.idTokenInfo = idTokenInfo;
+          }
+        } else {
+          // IdTokenInfo.status is one of Blocked, Expired, Invalid, NoCredit
+          // N.B. Other non-Accepted statuses should not be allowed to be stored.
+          transactionEventResponse.idTokenInfo = idTokenInfo;
+        }
+      } else {
+        // Assumed to always be valid without IdTokenInfo
+        transactionEventResponse.idTokenInfo = {
+          status: AuthorizationStatusEnumType.Accepted,
+          // TODO determine how/if to set personalMessage
+        };
+      }
+    }
+
+    if (
+      transactionEvent.eventType === TransactionEventEnumType.Started &&
+      transactionEventResponse &&
+      transactionEventResponse.idTokenInfo?.status ===
+        AuthorizationStatusEnumType.Accepted &&
+      transactionEvent.idToken
+    ) {
+      if (this._costUpdatedInterval) {
+        this._updateCost(
+          stationId,
+          transactionId,
+          this._costUpdatedInterval,
+          message.context.tenantId,
+        );
+      }
+
+      // TODO there should only be one active transaction per evse of a station.
+      // old transactions should be marked inactive and an alert should be raised (this can only happen in the field with charger bugs or missed messages)
+
+      // Check for ConcurrentTx
+      return await this._transactionEventRepository
+        .readAllActiveTransactionsByIdToken(transactionEvent.idToken)
+        .then((activeTransactions) => {
```

Comment:
Minor: it would be perfect if we can get rid of the Promise.then here as well.

Reply:
Done!